public/pages/workflow_detail/workflow_inputs/processors_list.tsx

```diff
@@ -52,30 +62,188 @@ const PANEL_ID = 0;
  */
 export function ProcessorsList(props: ProcessorsListProps) {
   const { values } = useFormikContext<WorkflowFormValues>();
-
-  // Processor added state. Used to automatically open accordion when a new
-  // processor is added, assuming users want to immediately configure it.
+  const [version, setVersion] = useState<string>('');
+  const location = useLocation();
   const [processorAdded, setProcessorAdded] = useState<boolean>(false);
-
-  // Popover state when adding new processors
   const [isPopoverOpen, setPopover] = useState(false);
+  const [processors, setProcessors] = useState<IProcessorConfig[]>([]);
+
   const closePopover = () => {
     setPopover(false);
   };
 
-  // Current processors state
-  const [processors, setProcessors] = useState<IProcessorConfig[]>([]);
+  const handlePopoverClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  useEffect(() => {
+    const dataSourceId = getDataSourceFromURL(location).dataSourceId;
+    if (dataSourceId) {
+      getEffectiveVersion(dataSourceId)
+        .then((ver) => {
+          setVersion(ver);
+        })
+        .catch(console.error);
+    }
+  }, [location]);
+
   useEffect(() => {
-    if (props.uiConfig && props.context) {
-      setProcessors(
-        props.context === PROCESSOR_CONTEXT.INGEST
-          ? props.uiConfig.ingest.enrich.processors
-          : props.context === PROCESSOR_CONTEXT.SEARCH_REQUEST
-          ? props.uiConfig.search.enrichRequest.processors
-          : props.uiConfig.search.enrichResponse.processors
-      );
+    const loadProcessors = async () => {
+      if (props.uiConfig && props.context) {
+        let currentProcessors =
+          props.context === PROCESSOR_CONTEXT.INGEST
+            ? props.uiConfig.ingest.enrich.processors
+            : props.context === PROCESSOR_CONTEXT.SEARCH_REQUEST
+            ? props.uiConfig.search.enrichRequest.processors
+            : props.uiConfig.search.enrichResponse.processors;
+
+        // If version is 2.17 or 2.18, filter out ML processors
+        if (
+          version &&
+          semver.gte(version, '2.17.0') &&
+          semver.lt(version, '2.19.0')
+        ) {
+          currentProcessors =
+            currentProcessors?.filter(
+              (processor) => processor.type !== 'ml_inference'
+            ) || [];
+        }
+        setProcessors(currentProcessors || []);
+        props.onProcessorsChange?.(currentProcessors?.length || 0);
+      }
+    };
+
+    loadProcessors();
+  }, [props.context, props.uiConfig, version]);
+
+  const getMenuItems = () => {
```

Comment:
Nice! This fn lgtm, looks like all of the states covered. It was about time to throw the growing list of processors into somewhat of a more structured origin anyways :)